### PR TITLE
Bugfix: prevent negative value for bribes

### PR DESCRIPTION
**Pull request by Hurleveur**
**Bugfix:** This PR addresses not being able to land with bribes when your value is negative (with the shooting star having a negative cost)

## Fix Details
Just make sure the total value is at least 1, so we dont have negative bribe with sqrt of a negative int going on.

## Testing Done
this PR with that savefile (you can pay them to land just fine)
image
master:
image
which you can't even pay
image

## Save File
[Isatrane_Reck.txt](https://github.com/endless-sky/endless-sky/files/11872182/Isatrane_Reck.txt)

### CODE REVIEW

File: source/HailPanel.cpp [hunk 1]
```diff
@@ -350,7 +350,10 @@ void HailPanel::SetBribe(double scale)
 	for(const shared_ptr<Ship> &it : player.Ships())
 		value += it->Cost();
 
+	if(value < 0)
+		value = 1;
+
 	bribe = 1000 * static_cast<int64_t>(sqrt(value) * scale);
-	if(scale && !bribe)
+	if(scale && bribe)
```
Comment (RisingLeaf) [inline source/HailPanel.cpp:357]: but why this change?

Comment (Hurleveur) [reply source/HailPanel.cpp:357]:
```suggestion
	if(scale && !bribe)
```

**Review by RisingLeaf — APPROVED** [review]

File: source/HailPanel.cpp [hunk 2]
```diff
@@ -350,6 +350,9 @@ void HailPanel::SetBribe(double scale)
 	for(const shared_ptr<Ship> &it : player.Ships())
 		value += it->Cost();
 
+	if(value < 0)
```
Comment (quyykk) [inline source/HailPanel.cpp:353]: Should this be `<=`?

Comment (Hurleveur) [reply source/HailPanel.cpp:353]: Doesnt matter because if its 0 the code after will make it 1000 (and I think its there for that 0 case? Mb put <= and remove it?)

Comment (Hurleveur) [reply source/HailPanel.cpp:353]:
```suggestion
	if(value <= 0)
```

File: source/HailPanel.cpp [hunk 3]
```diff
@@ -350,6 +350,9 @@ void HailPanel::SetBribe(double scale)
...
+		value = 1;
+
 	bribe = 1000 * static_cast<int64_t>(sqrt(value) * scale);
 	if(scale && !bribe)
```
Comment (Hurleveur) [inline source/HailPanel.cpp:357]:
```suggestion
```

File: source/HailPanel.cpp [hunk 4]
```diff
@@ -350,6 +350,9 @@ void HailPanel::SetBribe(double scale)
...
 		bribe = 1000;
```
Comment (Hurleveur) [inline source/HailPanel.cpp:358]:
```suggestion
```

**Review by quyykk — APPROVED** [review]
